### DISCUSSION TITLE
OCILIB: Get Index of a Bind variable

### DISCUSSION THREAD

**Discussion by aravindsreeram062** [vrogier/ocilib#280] [Q&A] [answered]
I'm writing a C code to get index of a variable using 

> OCI_GetBindIndex() 

but its returning always 0 as index. could anyone help me how to fetch the correct index using bind variable

```
big_int l_result = 0;
big_int l_emp_id= 1234;
session_query = "BEGIN :l_result := emp.fetch_salary(:l_emp_id); END;";

OCI_BindBigInt(st, ":l_result", &l_result);
OCI_BindBigInt(st, ":l_emp_id", &l_emp_id);

printf("Index is <%d>\n",OCI_GetBindIndex(st,"l_result"));
printf("Index is <%d>\n", OCI_GetBindIndex(st, "l_emp_id"));
```

The index is 0 always.

**Comment by vrogier** [comment, marked as answer]
Hi,

You need to pass to OCI_GetBindIndex() the same bind name used with OCI_Bindxxx(), e.g.  ":l_result" and ":l_emp_id".

Regards,

Vincent

**Reply by aravindsreeram062** [reply]
thank you so much :) it worked
